bentoml/_internal/bento/bento.py

```diff
@@ -82,21 +82,30 @@ def get_default_bento_readme(svc: "Service"):
 @attr.define(repr=False, auto_attribs=False)
 class Bento(StoreItem):
     _tag: Tag = attr.field()
-    _fs: "FS" = attr.field()
+    __fs: "FS" = attr.field()
 
-    info: "BentoInfo"
+    _info: "BentoInfo"
 
     _model_store: ModelStore
     _doc: t.Optional[str] = None
 
-    @_fs.validator
+    _flushed: bool = False
+
+    @staticmethod
+    def _export_ext() -> str:
+        return "bento"
+
+    @__fs.validator
     def check_fs(self, _attr: t.Any, new_fs: "FS"):
-        new_fs.makedir("models", recreate=True)
-        self._model_store = ModelStore(new_fs.opendir("models"))
+        try:
+            new_fs.makedir("models", recreate=True)
+            self._model_store = ModelStore(new_fs.opendir("models"))
+        except fs.errors.ResourceReadOnly:
```

Comment:
just curious, what's the reason for the `fs.errors.ResourceReadOnly` check here?

Reply:
If the underlying FS is read-only (e.g. if the user imported a tarfile) the `makedir` call will fail but we still want to try to open the model store.

Reply:
Let's add an inline comment here?